code/modules/projectiles/guns/projectile/improvised.dm

```diff
@@ -17,6 +17,15 @@
 	origin_tech = "combat=2;materials=2"
 	handle_casings = CYCLE_CASINGS
 	load_method = SINGLE_CASING
+	
+/obj/item/weapon/gun/projectile/shotgun/improvised/special_check(var/mob/living/carbon/human/M)
+	if(prob(40 - (loaded.len * 10)))
+		M << "<span class='danger'>[src] blows up in your face.</span>"
+		M.take_organ_damage(0,20)
+		M.drop_item()
+		del(src)

```

Comment:
Bad del command.

`qdel(src)` is to be used in all cases, other than save files, SQLLite files, and I forget one more instance maybe.